src/core/DY.Core.Trace.Base.fst

```diff
@@ -254,6 +254,20 @@ val event_at_grows:
   [SMTPat (event_at tr1 i e); SMTPat (tr1 <$ tr2)]
 let event_at_grows #label_t tr1 tr2 i e = ()
 
+val last_entry_exists:
+  #label_t:Type ->
+  tr:trace_ label_t ->
+  Lemma
+    (requires Snoc? tr )
+    (ensures (
+       let Snoc _ last = tr in
+       event_exists tr last
+    ))
+    [SMTPat (Snoc? tr)]
```

Comment:
I am genuinely surprised about this SMT pattern, it works without having to write `assert(Snoc? tr)`?!

Reply:
It works well in `Snoc` cases of `match`es.
Do you suggest a different pattern?

Reply:
No, if it works well that is fine! I never wrote a pattern like that so I didn't know how it would behave.